test/unit/lifecycle/properties.js

```diff
@@ -455,6 +455,80 @@ describe('lifecycle/property', () => {
     });
 
     describe('set()', () => {
+      describe('for elements initialised from the DOM', () => {
+        it('is called for an attribute prop when initialised as an attribute', (done) => {
+          const { skate, safe } = element();
+          let calls = 0;
+
+          skate({
+            props: {
+              foo: {
+                attribute: true,
+                set (elem, data) {
+                  expect(data.newValue).to.equal('bar');
+                  expect(data.oldValue).to.equal(null);
+                  ++calls;
+                }
+              }
+            }
+          });
+
+          afterMutations(
+            () => fixture(`<${safe} foo="bar" />`),
```

Comment:
It is magic. #891 is for refactoring that out and making this a bit simpler using just `define()`.